draft-ietf-quic-load-balancers.md

```diff
@@ -752,20 +767,108 @@ such as congestion window, RTT, or MTU. Opaque data MAY also allow servers to
+
+* The associated data is a formatted as a pseudo header by 
+  combining the clear-text part of the token with the IP address
+  of the client.
+~~~~~
+0                   1                   2                   3
+0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                                                               +
+|                                                               |
++                        IP Address (128)                       +
+|                                                               |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                     Server ID (32)                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                 Token sequence number (64)                    +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+| Key seq.(8)   |
++-+-+-+-+-+-+-+-+
+~~~~~
+* The input plaintext for the AEAD is the token body. The output ciphertext
+of the AEAD is transmitted in place of token body.
+* The AEAD Checksum is computed as part of the AEAD encryption process,
+and is verified during decryption.
+
 ### Configuration Agent Actions
 
-The configuration agent generates and distributes a "token key" and a list of
+The configuration agent generates and distributes a "token key", a "token IV",
+a key sequence,
+and a list of
 QUIC versions the service supports. It must also inform the service if a NAT
 lies between the service and the servers.
 
+The configuration also specifies whether the unique token number is
+generated using the "random" or the "controlled" methods. If the controlled
+method is used, the configuration agent provisions each server with a
+unique 32 bit server ID.
+
 ### Service Requirements
 
 When in active mode, the service MUST generate Retry tokens with the format
 described above when it receives a client Initial packet with no token.
 
-In active mode, the service SHOULD decrypt the first 16 octets of incoming
-tokens. The service SHOULD drop packets with an IP address that does not match,
-and SHOULD forward packets that do, regardless of the other fields.
+In active mode, the service SHOULD decrypt the incoming
+tokens. The service SHOULD drop packets with unknown key sequence,
+or an AEAD checksum that does not match the expected value.
+(By construction, the AEAD checksum will only match is the client
+IP Address also matches.)
+It SHOULD forward packets that do, regardless of the other fields.
+
+DISCUSS: should we say something about the time stamp?
```

Comment:
Why wouldn't it? As there are multiple token generators in this model, the time thresholds are going to have be forgiving about clock skew anyway.

Reply:
Result of call: yes, the service should look at the timestamp (with allowances for clock skew)